packages/playwright-core/src/client/network.ts

```diff
@@ -142,7 +157,9 @@ export class Request extends ChannelOwner<channels.RequestChannel> implements ap
    * @deprecated
    */
   headers(): Headers {
-    return this._provisionalHeaders.headers();
+    if (this._fallbackOverrides.headers)
```

Comment:
Remove the `if` block , `return this._fallbackOverrides.headers || this._provisionalHeaders.headers();` should be sufficient

Reply:
I need to normalize those (lower case names, etc)